server/conditions/tests/conditions_test.go

```diff
@@ -550,3 +550,120 @@ func TestCollisionWithBottomLeftCorner(t *testing.T) {
 	assert.Equal(t, 3, nextState.Players[0].Facing)
 	assert.Equal(t, gameState.Players[0].Health-2, nextState.Players[0].Health)
 }
+
+/*
++++
+++B
+++A
+*/
+func TestEdgeCollisionWithPlayersGoingSameDirectionPartI(t *testing.T) {
```

Comment:
These three tests are all dumbed down versions of the state and inputs that caused #65.  Then I made some slight variations to the inputs to make sure the solution was robust.